### ISSUE TITLE
Weird training results with RoBERTa

### ISSUE THREAD

**Issue by duyvuleo** [pytorch/fairseq#1630]
Hi,

I managed to train a RoBERTa version for another language other than English. My crawled data (news, wiki, subtitle, ...) include a split of ~173M sentences for train, 26K sentences for valid/test. 

After training for several hours (~55K updates), I got the following loss/ppl results which look weird to me. 
![loss](https://user-images.githubusercontent.com/5590702/72643835-ae115980-39c3-11ea-9c52-cedbc6aebf9f.png)
![ppl](https://user-images.githubusercontent.com/5590702/72643872-c08b9300-39c3-11ea-90f5-3313aeaa74e8.png)

My configuration for training RoBERTa is as follows:
```TOTAL_UPDATES=1000000    # Total number of training steps
WARMUP_UPDATES=10000    # Warmup the learning rate over this many updates
PEAK_LR=0.0005          # Peak learning rate, adjust as needed
TOKENS_PER_SAMPLE=512   # Max sequence length
MAX_POSITIONS=512       # Num. positional embeddings (usually same as above)
MAX_SENTENCES=16        # Number of sequences per batch (batch size)
UPDATE_FREQ=16          # Increase the batch size 16x

DATA_DIR=/data-bin
MODEL_DIR=/models/full

mkdir $MODEL_DIR/tensorboard_log

fairseq-train --fp16 $DATA_DIR \
    --task masked_lm --criterion masked_lm \
    --arch roberta_base --sample-break-mode eos --tokens-per-sample $TOKENS_PER_SAMPLE --skip-invalid-size-inputs-valid-test \
    --optimizer adam --adam-betas '(0.9,0.98)' --adam-eps 1e-6 --clip-norm 0.0 \
    --lr-scheduler polynomial_decay --lr $PEAK_LR --warmup-updates $WARMUP_UPDATES --total-num-update $TOTAL_UPDATES \
    --dropout 0.1 --attention-dropout 0.1 --weight-decay 0.01 \
    --max-sentences $MAX_SENTENCES --update-freq $UPDATE_FREQ \
    --max-update $TOTAL_UPDATES --log-format simple --log-interval 1 \
    --max-epoch 100 \
    --validate-interval 1 \
    --patience 5 \
    --save-dir $MODEL_DIR \
    --save-interval-updates 10000 \
    --keep-interval-updates 3 \
    --keep-last-epochs 3 \
    --tensorboard-logdir $MODEL_DIR/tensorboard_log```

Would you please help advise what went wrong for my training? Thanks!

**Comment by myleott** [comment]
Most likely the learning rate is a bit too large or the batch size is too small. What is the batch size?

You can try resuming training from a checkpoint a bit before it explodes and use a different `--seed` (which will shuffle the data order and may avoid the explosion). Alternatively you may need to reduce the learning rate.

**Comment by duyvuleo** [comment]
Thanks for your quick response. The batch_size is 2048 (MAX_SENTENCES * UPDATE_FREQ * 8 GPUs?).  I'll try you suggestion.
